src/watch.ts

```diff
@@ -110,7 +111,19 @@ export class BuildRunnerWatch {
     if (this.process !== undefined) {
       console.log('Going to try to kill the watch...');
       console.log('PID of parent is ' + this.process.pid);
-      const pid = cp.getChildPID(this.process)!;
+      let pid: string | undefined = cp.getChildPID(this.process?.pid?.toString());
+      // when using flutter, the pid tree is like this:
+      //  \-+- ${process.pid}  bash <flutter_path>/bin/flutter pub run build_runner watch
+      //   \-+- child_pid1      <flutter_path>/bin/cache/dart-sdk/bin/dart --disable-dart-dev --packages=<flutter_path>/packages/flutter_tools/.packages <flutter_path>/bin/cache/flutter_tools.snapshot pub run build_runner watch
+      //    \--- child_pid2      <flutter_path>/bin/cache/dart-sdk/bin/dart __deprecated_pub run build_runner watch
+      // we thus need to kill child_pid2 by finding process.pid's grandchild
+      if (this.command === "flutter") {
+        const newPid = cp.getChildPID(pid);
+        if (newPid !== undefined) {
+          pid = newPid;
+        }
+      }
+
```

Comment:
After some research it looks like when using flutter, it actually runs the dart command proxied so we need to go one depth deeper and find the grandchild's pid:
![image](https://user-images.githubusercontent.com/25441359/145888289-3c324d01-8e6a-43bf-919d-17b2f0b11aa5.png)
Whereas on dart: 
![image](https://user-images.githubusercontent.com/25441359/145888394-b528cef1-52f3-48d9-b743-6c90753a8a67.png)